src/_06_Intro_To_Hash_Maps/_02_LogSearch.java

```diff
@@ -1,33 +1,137 @@
-	 * 				When this button is clicked, prompt the user to enter an ID using an input dialog.
-	 * 				If this ID exists in the HashMap, remove it. Otherwise, notify the user that the ID
-	 * 				is not in the list. 
+	 * When this is complete, add a fourth button to your window. Button 4: Remove
+	 * Entry When this button is clicked, prompt the user to enter an ID using an
+	 * input dialog. If this ID exists in the HashMap, remove it. Otherwise, notify
+	 * the user that the ID is not in the list.
 	 *
-	 * */
+	 */
+
+	private HashMap<String, String> listOfStudents = new HashMap<>();
+	String studentsAndIDs;
+
+	// Main UI variables
+	private JFrame mainUIFrame = new JFrame();
+	private JPanel mainUIPanel = new JPanel();
+	private JButton addEntryButton = new JButton();
+	private JButton searchByIDButton = new JButton();
+	private JButton viewListButton = new JButton();
+
+	// Add an entry UI variables
+	private JFrame addEntryUIFrame = new JFrame();
+	private JPanel addEntryUIPanel = new JPanel();
+	private JLabel IDLabel = new JLabel("ID: ");
+	private JTextField IDTextField = new JTextField(15);
+	private JLabel nameLabel = new JLabel("Name: ");
+	private JTextField nameTextField = new JTextField(15);
+	private JButton addNewEntryButton = new JButton();
+
+	// View list UI variables
+	private JFrame viewListUIFrame = new JFrame();
+	private JPanel viewListUIPanel = new JPanel();
+	private JTextArea listOfStudentsTextArea = new JTextArea();
+	
+	public void createUI() {
+		mainUIFrame.setVisible(true);
+		mainUIFrame.add(mainUIPanel);
+		mainUIPanel.add(addEntryButton);
+		addEntryButton.setText("Add an Entry");
+		addEntryButton.addActionListener(this);
+		mainUIPanel.add(searchByIDButton);
+		searchByIDButton.setText("Search By ID");
+		searchByIDButton.addActionListener(this);
+		mainUIPanel.add(viewListButton);
+		viewListButton.setText("View List");
+		viewListButton.addActionListener(this);
+		mainUIFrame.pack();
+		mainUIFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+	}
+
+	public void addAddEntryUI() {
+		addEntryUIFrame.setVisible(true);
+		addEntryUIFrame.add(addEntryUIPanel);
+		addEntryUIPanel.add(IDLabel);
+		addEntryUIPanel.add(IDTextField);
+		addEntryUIPanel.add(nameLabel);
+		addEntryUIPanel.add(nameTextField);
+		addEntryUIPanel.add(addNewEntryButton);
+		addNewEntryButton.setText("Add Student");
+		addNewEntryButton.addActionListener(this);
+		addEntryUIFrame.pack();
+	}
 	
+	public void addViewListUI() {
+		viewListUIFrame.setVisible(true);
+		viewListUIFrame.add(viewListUIPanel);
+		viewListUIPanel.add(listOfStudentsTextArea);
+		listOfStudentsTextArea.setEditable(false);
+		viewListUIFrame.pack();
+	}
+
+	public static void main(String[] args) {
+		_02_LogSearch thisClass = new _02_LogSearch();
+		thisClass.createUI();
+	}
+
+	@Override
+	public void actionPerformed(ActionEvent e) {
+		JButton buttonClicked = (JButton) e.getSource();
+		if (buttonClicked == addEntryButton) {
+			this.addAddEntryUI();
+		}
+
```

Comment:
What's the difference between the addEntryButton and the addNewEntryButton...renaming them might help clarify the confusion❓